internal/fs/inode/name.go

```diff
@@ -123,3 +123,27 @@ func (name Name) IsDirectChildOf(parent Name) bool {
 	cleanDiff := strings.TrimSuffix(diff, "/")
 	return !strings.Contains(cleanDiff, "/")
 }
+
+// ParentName returns the Name of the parent directory of the current Name.
+func (name Name) ParentName() Name {
+	if name.IsBucketRoot() {
+		panic("Root has no parent")
```

Comment:
See if we can replace the panic with error. As we don't want to crash the whole app instead return input/output error for this particular operation.